### PR TITLE
Generate kotlin source files instead of java when "Use Kotlin" selected in gdx-setup

### PR DESCRIPTION
**Pull request by haxpor**
Most likely users want to work with pure Kotlin source files or as much as they can when they selected to use Kotlin for project in gdx-setup.

Notable changes are
* Add kotlin template files for all launcher classes, and core's main class. These are for project generation to use.
* Checking is based on `Language` for its `JAVA` and `KOTLIN`, and for HTML target, also check `gwtSupported` flag.
* Change kotlin version to `1.1.3` instead of `1.1.1` as there should be no major reason to hang back and use older version of kotlin.

**PS**. Actually there's 1 single commit that relevant to this changes. But due to I update my forked repo to latest libgdx after my [previous PR](https://github.com/libgdx/libgdx/pull/4741) gets merged. Advice is welcome on how to squeeze it to only 1 commit before merging.

### CODE REVIEW

File: extensions/gdx-setup/src/com/badlogic/gdx/setup/GdxSetup.java [hunk 1]
```diff
@@ -253,17 +253,30 @@ public void build (ProjectBuilder builder, String outputDir, String appName, Str
 		project.files.add(new ProjectFile("gradle/wrapper/gradle-wrapper.properties", false));
 		project.files.add(new ProjectFile("gradle.properties"));
 
-		// core project
+		// CoreGdxDefinition project
 		project.files.add(new ProjectFile("core/build.gradle"));
-		project.files.add(new ProjectFile("core/src/MainClass", "core/src/" + packageDir + "/" + mainClass + ".java", true));
+
+		if (language == Language.JAVA) {
```
Comment (Tom-Ski) [inline extensions/gdx-setup/src/com/badlogic/gdx/setup/GdxSetup.java]: I'd rather abstract this out now, have Language define a folder path for the src template files, and pick them up automatically from there.

Comment (haxpor) [inline extensions/gdx-setup/src/com/badlogic/gdx/setup/GdxSetup.java]: Ok, i'm done with abstracting resource out, and also fixed a problem regarding to running ios-moe in separate commit. Please check them out. Thanks!

**Comment by Namek** [comment]
@haxpor can you update and squash this PR?

**Comment by haxpor** [comment]
@Namek hey thanks for interest in this. I'm really not sure I can do this in the time being as it's a couple of years ago, and as I changed the working system now there's quite frictions to also update gradle project. Please feel free to extract and work on top of it. My apology I can't handle this myself.

**Comment by b3n10** [comment]
Is this an issue regarding project files not generating to kotlin but instead stays in java even "Use Kotlin" is selected in setup? This is what I'm currently experiencing on version 1.9.10

**Comment by MrStahlfelge** [comment]
@b3n10 please open an issue. I suggest to take a look at gdx-liftoff though.
Closing this PR as haxpor said he can't update it anyway.

**Pull request closed without merging**